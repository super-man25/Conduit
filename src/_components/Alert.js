// @flow
import * as React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import styled from 'styled-components';

import { colors, shadows, zIndexes } from '_constants';
import { actions } from '_state/alert';
import { Icon } from './Icon';
import { Flex } from './Flex';

const StyledApiAlert: React.ComponentType<{}> = styled(Flex)`
  position: absolute;
  right: 25px;
  top: 25px;
  z-index: ${zIndexes.BASE};
  width: 350px;
  background-color: ${colors.white};
  border-color: ${(props) =>
    props.type === 'apiError' ? colors.red : colors.green};
  box-shadow: ${shadows.SMALL};
  visibility: ${(props) => (props.show ? 'visible' : 'hidden')};
  cursor: pointer;
  padding: 25px;
`;

const StatusIcon: React.ComponentType<{}> = styled.div`
  margin-right: 15px;
`;

const StatusText: React.ComponentType<{}> = styled.div`
  font-size: 20px;
  font-weight: 300;
  letter-spacing: 1px;
  color: ${(props) => props.color};
  margin-bottom: 3px;
`;

const MsgText: React.ComponentType<{}> = styled.div`
  font-size: 13px;
  font-weight: 300;
`;

export const Alert = () => {
  const alertState = useSelector(({ alert }) => alert);
  const dispatch = useDispatch();
  const clearAlert = () => dispatch(actions.clear());

  const show = alertState.type !== null && alertState.message !== null;
  const statusColor =
    alertState.type === 'apiError' ? colors.red : colors.green;
  const statusText = alertState.type === 'apiError' ? 'Error:' : 'Success!';

  return (
    <StyledApiAlert {...alertState} show={show} onClick={clearAlert}>
      <StatusIcon>
        <Icon name={alertState.type} size={40} color={statusColor} />
      </StatusIcon>
      <Flex direction="column">
        <StatusText color={statusColor}>{statusText}</StatusText>
        <MsgText>{alertState.message}</MsgText>
      </Flex>
    </StyledApiAlert>
  );
};
