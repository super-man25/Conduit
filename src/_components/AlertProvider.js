import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { history } from '_helpers';
import { actions as alertActions } from '_state/alert';

class AlertProviderPresenter extends React.Component {
  constructor(props) {
    super(props);

    const { dispatch } = this.props;
    history.listen(() => {
      dispatch(alertActions.clear());
    });
  }

  render() {
    const { alert, children } = this.props;

    // TODO: style alerts
    return (
      <div>
        {alert &&
          alert.show && (
            <div className={`alert ${alert.type}`}>{alert.message}</div>
          )}
        {children}
      </div>
    );
  }
}

AlertProviderPresenter.propTypes = {
  /** Redux dispatch */
  dispatch: PropTypes.func,

  /** Alert object */
  alert: PropTypes.shape({
    message: PropTypes.string,
    type: PropTypes.string
  }),

  /** Children */
  children: PropTypes.node
};

function mapStateToProps(state) {
  return {
    alertState: state.alert
  };
}

export const AlertProvider = connect(mapStateToProps)(AlertProviderPresenter);
