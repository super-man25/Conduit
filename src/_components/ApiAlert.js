// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { cssConstants, shadows, zIndexes } from '_constants';
import { Icon } from './Icon';

type Props = {
  alertState: {
    type: 'api-error' | 'api-success',
    message: string
  }
};

export const ApiAlertDiv: React.ComponentType<{}> = styled.div`
  position: absolute;
  right: 24px;
  top: 24px;
  z-index: ${zIndexes.BASE};
  font-size: 18px;
  width: 357px;
  height: 90px;
  background-color: ${cssConstants.PRIMARY_WHITE};
  border-color: ${(props) =>
    props.type === 'api-error'
      ? cssConstants.SECONDARY_RED
      : cssConstants.SECONDARY_GREEN};
  box-shadow: ${shadows.SMALL};
  visibility: ${(props) => (props.show ? 'visible' : 'hidden')};
`;

const StatusIcon: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 26px;
  left: 26px;
`;

const StatusText: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 26px;
  left: 93px;
  font-size: 1.2em;
  font-weight: 300;
  letter-spacing: 1px;
  color: ${(props) => props.color};
`;

const MsgText: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 54px;
  left: 93px;
  font-size: 0.7em;
  font-weight: 300;
  margin-right: 24px;
  color: ${cssConstants.PRIMARY_BLACK};
`;

export class ApiAlertPresenter extends React.Component<Props> {
  render() {
    const { alertState } = this.props;
    const show = {
      show: alertState.type !== null && alertState.message !== null
    };
    const statusColor =
      alertState.type === 'api-error'
        ? cssConstants.SECONDARY_RED
        : cssConstants.SECONDARY_GREEN;
    const statusText = alertState.type === 'api-error' ? 'Error:' : 'Success!';

    return (
      <ApiAlertDiv {...alertState} {...show}>
        <StatusIcon>
          <Icon name={alertState.type} size={39} color={statusColor} />
        </StatusIcon>
        <StatusText color={statusColor}>{statusText}</StatusText>
        <MsgText>{alertState.message}</MsgText>
      </ApiAlertDiv>
    );
  }
}

function mapStateToProps(state) {
  return {
    alertState: state.alert
  };
}

export const ApiAlert = connect(
  mapStateToProps,
  null
)(ApiAlertPresenter);
