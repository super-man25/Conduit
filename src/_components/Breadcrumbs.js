import styled from 'styled-components';
import { cssConstants } from '../_constants';

export const Breadcrumbs = styled.div`
  width: 100%;
  height: 30px;
  margin: 0;
  padding: 0;
  background: ${cssConstants.PRIMARY_WHITE};
  font-size: 12px;
`;
