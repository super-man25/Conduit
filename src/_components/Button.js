// @flow

import * as React from 'react';
import styled from 'styled-components';
import { cssConstants, shadows } from '_constants';
import { withBoxModelProps } from '_helpers/style-utils';

type Props = {
  fontSize: string,
  textAlign: string,
  minWidth: string,
};

export const Button: React.ComponentType<Props> = withBoxModelProps(styled.button`
  border: none;
  min-width: ${(props) => props.minWidth || '100px'};
  padding: 1rem 2rem;
  text-align: ${(props) => props.textAlign || 'center'};
  border-radius: 3px;
  font-size: ${(props) => props.fontSize || '1rem'};
  cursor: pointer;
  transition-duration: 0.3s;

  &:disabled {
    cursor: not-allowed;
  }
`);

export const PrimaryButton = styled(Button)`
  transition: opacity 100ms ease-out;
  color: ${cssConstants.PRIMARY_WHITE};
  background-color: ${cssConstants.PRIMARY_BLUE};

  &:disabled {
    color: ${cssConstants.PRIMARY_GRAY};
    background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
    opacity: 0.5;
  }

  &:active {
    background-color: ${cssConstants.PRIMARY_LIGHT_BLUE};
    box-shadow: ${shadows.ACTIVE_BUTTON};
  }

  &:hover:not(:disabled) {
    background-color: ${cssConstants.PRIMARY_BLUE_HOVER};
  }
`;

export const SecondaryButton = styled(Button)`
  background-color: ${cssConstants.PRIMARY_WHITE};
  border: 2px solid ${cssConstants.PRIMARY_BLUE};
  color: ${cssConstants.PRIMARY_BLUE};
  transition: border linear;

  &:disabled {
    border: 2px solid ${cssConstants.PRIMARY_GRAY};
    color: ${cssConstants.PRIMARY_GRAY};
    opacity: 0.5;
  }

  &:active {
    background-color: ${cssConstants.SECONDARY_BACKGROUND_LIGHTEST_BLUE};
    color: ${cssConstants.PRIMARY_LIGHT_BLUE};
    outline: 2px solid ${cssConstants.PRIMARY_LIGHT_BLUE};
    box-shadow: ${shadows.ACTIVE_BUTTON};
  }

  &:hover:not(:disabled) {
    border: 3px solid ${cssConstants.PRIMARY_BLUE};
    background-color: ${cssConstants.SECONDARY_BACKGROUND_LIGHTEST_BLUE};
    color: ${cssConstants.PRIMARY_BLUE};
  }
`;

export const TertiaryButton = styled(Button)`
  background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
  opacity: 0.8;

  &:disabled {
    opacity: 0.5;
  }

  &:hover:not(:disabled) {
    opacity: 1;
    background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
  }
`;

export const TextButton = styled(Button)`
  color: ${cssConstants.PRIMARY_BLUE};
  background-color: inherit;

  &:disabled {
    color: ${cssConstants.PRIMARY_LIGHT_GRAY};
    opacity: 0.5;
  }

  &:hover:not(:disabled) {
    color: ${cssConstants.PRIMARY_LIGHT_BLUE};
  }
`;
