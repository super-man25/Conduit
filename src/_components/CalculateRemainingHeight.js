import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { ReactHeight } from 'react-height';

const Container = styled.div`
  height: 100%;
`;

export class CalculateRemainingHeight extends React.Component {
  constructor(props) {
    super(props);
    this.containerRef = React.createRef();
  }

  state = {
    remainingHeight: null
  };

  setRemainingHeight = (contentHeight) => {
    const containerNode = this.containerRef.current;
    this.setState({
      remainingHeight: containerNode.clientHeight - contentHeight
    });
  };

  render() {
    const children = [].concat(this.props.children);
    const { remainingHeight } = this.state;

    const total = children.length;

    const nodes = children.slice(0, total - 1);
    const func = children[total - 1];

    if (typeof func !== 'function') {
      throw new Error(
        'CalculateRemainingHeight: Expected the last child to be a function'
      );
    }

    return (
      <Container innerRef={this.containerRef}>
        <ReactHeight onHeightReady={this.setRemainingHeight}>
          {nodes}
        </ReactHeight>
        {remainingHeight !== 0 && func(remainingHeight)}
      </Container>
    );
  }
}

const childProp = PropTypes.oneOfType([PropTypes.node, PropTypes.func]);

CalculateRemainingHeight.propTypes = {
  /**
   * Children to display. The last child should be a function which receives the remaining
   * height in the container as a parameter and returns a node to render.
   */
  children: PropTypes.oneOfType([childProp, PropTypes.arrayOf(childProp)])
};
