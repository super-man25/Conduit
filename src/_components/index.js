export * from './Alert';
export * from './AsyncButton';
export * from './Breadcrumbs';
export * from './Button';
export * from './Center';
export * from './CenteredLoader';
export * from './ChartContainer';
export * from './ChartLegend';
export * from './Checkbox';
export * from './ChipButton';
export * from './ContentWrapper';
export * from './DatePicker';
export * from './DateRangeDropdown';
export * from './DateRangeFilter';
export * from './Dropdown';
export * from './DropdownButton';
export * from './EDLink';
export * from './EDText';
export * from './Flex';
export * from './Form';
export * from './FormInput';
export * from './FormInputArray';
export * from './HelpBlockDiv';
export * from './HorizontalBars';
export * from './Icon';
export * from './Icon/SVGIcons';
export * from './ImageLayout';
export * from './Input';
export * from './Integration';
export * from './Layout';
export * from './LeftNav';
export * from './LegendItem';
export * from './Loader';
export * from './MailtoLink';
export * from './MainContent';
export * from './Modal';
export * from './Panel';
export * from './ReportDownloadButton';
export * from './ScreenTitleBlock';
export * from './ScrollableList';
export * from './SecuredRoute';
export * from './SelectBox';
export * from './SelectDropdown';
export * from './Setting';
export * from './SettingButton';
export * from './SiteHeader';
export * from './SortableButton';
export * from './Spacing';
export * from './StyledTags';
export * from './Table';
export * from './Toggle';
export * from './UserWelcome';
