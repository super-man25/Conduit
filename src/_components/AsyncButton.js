// @flow

import * as React from 'react';
import { cssConstants } from '_constants';
import { PrimaryButton, Loader } from '_components';

type Props = {
  isLoading: boolean,
  disabled?: boolean,
  children: React.Node,
};

export const AsyncButton = (props: Props) => {
  const { isLoading, children, ...rest } = props;
  return (
    <PrimaryButton {...rest}>
      {props.isLoading ? (
        <Loader small color={cssConstants.PRIMARY_WHITE} />
      ) : (
        props.children
      )}
    </PrimaryButton>
  );
};
