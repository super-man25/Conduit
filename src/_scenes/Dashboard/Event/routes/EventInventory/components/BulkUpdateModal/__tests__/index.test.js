import React from 'react';
import { shallow } from 'enzyme';
import { BulkUpdateModalPresenter, selectActions } from '..';
import { SecondaryButton, PrimaryButton } from '_components';

describe('<BulkUpdateModal />', () => {
  const props = {
    rows: [{ seats: [1, 2, 3] }, { seats: [4, 5, 6] }, { seats: [7, 8, 9] }],
    loading: false,
    cancelBulkUpdate: jest.fn(),
    submitBulkUpdate: jest.fn()
  };

  it('should render correctly', () => {
    const wrapper = shallow(<BulkUpdateModalPresenter {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when loading is true', () => {
    const wrapper = shallow(
      <BulkUpdateModalPresenter {...props} loading={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when an invalid manual price is entered and blurred', () => {
    const wrapper = shallow(<BulkUpdateModalPresenter {...props} />);
    wrapper.setState({
      touched: {
        price: true
      },
      value: '12.222'
    });
    expect(wrapper.instance().isValidManualPrice).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when the selectedAction is changed', () => {
    const wrapper = shallow(
      <BulkUpdateModalPresenter {...props} loading={true} />
    );
    wrapper.setState({
      selectedAction: selectActions[1]
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('submitEnabled should be false if invalid information is set', () => {
    const wrapper = shallow(<BulkUpdateModalPresenter {...props} />);
    wrapper.setState({ value: 'abc' });
    expect(wrapper.instance().submitEnabled()).toBe(false);
  });

  it('submitEnabled should be true if valid information is set', () => {
    const wrapper = shallow(<BulkUpdateModalPresenter {...props} />);
    wrapper.setState({ value: '12.00' });
    expect(wrapper.instance().submitEnabled()).toBe(true);

    wrapper.setState({ selectedAction: selectActions[1], value: false });
    expect(wrapper.instance().submitEnabled()).toBe(true);
  });

  it('should call cancelBulkUpdate when the cancel button is clicked', () => {
    const fn = jest.fn();
    const wrapper = shallow(
      <BulkUpdateModalPresenter {...props} cancelBulkUpdate={fn} />
    );

    wrapper.find(SecondaryButton).simulate('click');
    expect(fn).toBeCalled();
  });

  it('should call submitBulkUpdate when the submit button is clicked', () => {
    const fn = jest.fn();
    const wrapper = shallow(
      <BulkUpdateModalPresenter {...props} submitBulkUpdate={fn} />
    );
    wrapper.setState({
      value: '12.22'
    });

    wrapper.find(PrimaryButton).simulate('click');
    expect(fn).toBeCalled();
  });
});
