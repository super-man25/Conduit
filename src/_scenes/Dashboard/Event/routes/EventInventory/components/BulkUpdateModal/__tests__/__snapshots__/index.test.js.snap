// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BulkUpdateModal /> should render correctly 1`] = `
<Styled(styled.div)>
  <styled.div>
    <styled.div>
      <Styled(Styled(styled.h4))>
        Bulk Inventory Update
      </Styled(Styled(styled.h4))>
      <Styled(styled.p)
        size={14}
        weight={300}
      >
        Updating inventory for 
        3
         row(s).
      </Styled(styled.p)>
    </styled.div>
    <styled.div>
      <Styled(styled.div)
        marginBottom="2rem"
      >
        <styled.label>
          Action
        </styled.label>
        <SelectDropdown
          full={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Update Manual Price",
                "value": 0,
              },
              Object {
                "label": "Update Pricing Status",
                "value": 1,
              },
            ]
          }
          placeholder="Select"
          selected={
            Object {
              "label": "Update Manual Price",
              "value": 0,
            }
          }
        />
      </Styled(styled.div)>
      <Styled(Styled(styled.div))
        marginBottom="2rem"
      >
        <styled.label>
          Manual Price
        </styled.label>
        <NumberInputField
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-fjmCvl",
                "isStatic": true,
                "rules": Array [
                  "
  background-color: ",
                  "#FFFFFF",
                  ";
  border: 1px solid ",
                  "#999999",
                  ";
  border-radius: 3px;
  height: 40px;
  margin: 0;
  padding: 0 0.5rem;
  width: 100%;
",
                ],
              },
              "displayName": "styled.input",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-fjmCvl",
              "target": "input",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          id="manual-price"
          name="price"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="$ Manual Price"
          type="number"
          value=""
        />
      </Styled(Styled(styled.div))>
      <Flex
        justify="flex-end"
      >
        <Styled(Styled(styled.button))
          margin="0 1rem 0 0"
          onClick={[MockFunction]}
          small={true}
        >
          Cancel
        </Styled(Styled(styled.button))>
        <AsyncButton
          disabled={false}
          isLoading={false}
          onClick={[Function]}
        >
          Submit
        </AsyncButton>
      </Flex>
    </styled.div>
  </styled.div>
  <styled.div
    onClick={[MockFunction]}
  />
</Styled(styled.div)>
`;

exports[`<BulkUpdateModal /> should render correctly when an invalid manual price is entered and blurred 1`] = `
<Styled(styled.div)>
  <styled.div>
    <styled.div>
      <Styled(Styled(styled.h4))>
        Bulk Inventory Update
      </Styled(Styled(styled.h4))>
      <Styled(styled.p)
        size={14}
        weight={300}
      >
        Updating inventory for 
        3
         row(s).
      </Styled(styled.p)>
    </styled.div>
    <styled.div>
      <Styled(styled.div)
        marginBottom="2rem"
      >
        <styled.label>
          Action
        </styled.label>
        <SelectDropdown
          full={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Update Manual Price",
                "value": 0,
              },
              Object {
                "label": "Update Pricing Status",
                "value": 1,
              },
            ]
          }
          placeholder="Select"
          selected={
            Object {
              "label": "Update Manual Price",
              "value": 0,
            }
          }
        />
      </Styled(styled.div)>
      <Styled(Styled(styled.div))
        marginBottom="2rem"
      >
        <styled.label>
          Manual Price
        </styled.label>
        <NumberInputField
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-fjmCvl",
                "isStatic": true,
                "rules": Array [
                  "
  background-color: ",
                  "#FFFFFF",
                  ";
  border: 1px solid ",
                  "#999999",
                  ";
  border-radius: 3px;
  height: 40px;
  margin: 0;
  padding: 0 0.5rem;
  width: 100%;
",
                ],
              },
              "displayName": "styled.input",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-fjmCvl",
              "target": "input",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          id="manual-price"
          name="price"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="$ Manual Price"
          type="number"
          value="12.222"
        />
        <Styled(Styled(styled.p))
          color="#DC0A0A"
          marginTop="0.5rem"
          size={12}
          weight={300}
        >
          Manual Price must be a valid dollar amount.
        </Styled(Styled(styled.p))>
      </Styled(Styled(styled.div))>
      <Flex
        justify="flex-end"
      >
        <Styled(Styled(styled.button))
          margin="0 1rem 0 0"
          onClick={[MockFunction]}
          small={true}
        >
          Cancel
        </Styled(Styled(styled.button))>
        <AsyncButton
          disabled={true}
          isLoading={false}
          onClick={[Function]}
        >
          Submit
        </AsyncButton>
      </Flex>
    </styled.div>
  </styled.div>
  <styled.div
    onClick={[MockFunction]}
  />
</Styled(styled.div)>
`;

exports[`<BulkUpdateModal /> should render correctly when loading is true 1`] = `
<Styled(styled.div)>
  <styled.div>
    <styled.div>
      <Styled(Styled(styled.h4))>
        Bulk Inventory Update
      </Styled(Styled(styled.h4))>
      <Styled(styled.p)
        size={14}
        weight={300}
      >
        Updating inventory for 
        3
         row(s).
      </Styled(styled.p)>
    </styled.div>
    <styled.div>
      <Styled(styled.div)
        marginBottom="2rem"
      >
        <styled.label>
          Action
        </styled.label>
        <SelectDropdown
          full={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Update Manual Price",
                "value": 0,
              },
              Object {
                "label": "Update Pricing Status",
                "value": 1,
              },
            ]
          }
          placeholder="Select"
          selected={
            Object {
              "label": "Update Manual Price",
              "value": 0,
            }
          }
        />
      </Styled(styled.div)>
      <Styled(Styled(styled.div))
        marginBottom="2rem"
      >
        <styled.label>
          Manual Price
        </styled.label>
        <NumberInputField
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-fjmCvl",
                "isStatic": true,
                "rules": Array [
                  "
  background-color: ",
                  "#FFFFFF",
                  ";
  border: 1px solid ",
                  "#999999",
                  ";
  border-radius: 3px;
  height: 40px;
  margin: 0;
  padding: 0 0.5rem;
  width: 100%;
",
                ],
              },
              "displayName": "styled.input",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-fjmCvl",
              "target": "input",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          id="manual-price"
          name="price"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="$ Manual Price"
          type="number"
          value=""
        />
      </Styled(Styled(styled.div))>
      <Flex
        justify="flex-end"
      >
        <Styled(Styled(styled.button))
          margin="0 1rem 0 0"
          onClick={[MockFunction]}
          small={true}
        >
          Cancel
        </Styled(Styled(styled.button))>
        <AsyncButton
          disabled={false}
          isLoading={true}
          onClick={[Function]}
        >
          Submit
        </AsyncButton>
      </Flex>
    </styled.div>
  </styled.div>
  <styled.div
    onClick={[MockFunction]}
  />
</Styled(styled.div)>
`;

exports[`<BulkUpdateModal /> should render correctly when the selectedAction is changed 1`] = `
<Styled(styled.div)>
  <styled.div>
    <styled.div>
      <Styled(Styled(styled.h4))>
        Bulk Inventory Update
      </Styled(Styled(styled.h4))>
      <Styled(styled.p)
        size={14}
        weight={300}
      >
        Updating inventory for 
        3
         row(s).
      </Styled(styled.p)>
    </styled.div>
    <styled.div>
      <Styled(styled.div)
        marginBottom="2rem"
      >
        <styled.label>
          Action
        </styled.label>
        <SelectDropdown
          full={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Update Manual Price",
                "value": 0,
              },
              Object {
                "label": "Update Pricing Status",
                "value": 1,
              },
            ]
          }
          placeholder="Select"
          selected={
            Object {
              "label": "Update Pricing Status",
              "value": 1,
            }
          }
        />
      </Styled(styled.div)>
      <Styled(Styled(styled.div))
        marginBottom="1.5rem"
      >
        <styled.label>
          Pricing
        </styled.label>
        <Toggle
          isChecked=""
          onChange={[Function]}
          size="small"
        />
      </Styled(Styled(styled.div))>
      <Flex
        justify="flex-end"
      >
        <Styled(Styled(styled.button))
          margin="0 1rem 0 0"
          onClick={[MockFunction]}
          small={true}
        >
          Cancel
        </Styled(Styled(styled.button))>
        <AsyncButton
          disabled={false}
          isLoading={true}
          onClick={[Function]}
        >
          Submit
        </AsyncButton>
      </Flex>
    </styled.div>
  </styled.div>
  <styled.div
    onClick={[MockFunction]}
  />
</Styled(styled.div)>
`;
